spec/formatting.md

```diff
@@ -489,6 +489,15 @@ Earlier _selectors_ in the _matcher_'s list of _selectors_ have a higher priorit
 When all of the _selectors_ have been processed,
 the earliest-sorted _variant_ in the remaining list of _variants_ is selected.
 
+> [!NOTE]
+> A _selector_ is not a _declaration_.
+> Even when the same _function_ can be used for both formatting and selection
+> of a given _operand_
+> the _annotation_ that appears in a _selector_ has no effect on subsequent
+> _selectors_ nor on the formatting used in _placeholders_.
```

Comment:
```suggestion
> [!NOTE]
> _Expressions_, including _expressions_ that appear as _selectors_,
> have no side effects and are not _declarations_.
> That is, they do not introduce or override _variable_ names.
> This means that the _annotation_ of a _variable_ in a _selector_
> does not change what is bound to the _variable_,
> even if the same _function_ can be used for both formatting and selection
> of a given _operand_.
> The meaning of a _variable_ reference that occurs within a _variant_
> depends only on the explicit or implicit _declaration_ of that _variable_.
```

Just a suggestion. I think this text is difficult to understand outside the context of the discussions we've been having, and without an example. The same might be true of my suggested text. I guess if it's confusing, people will ask about it during the tech preview feedback period.

Reply:
The right hand size of a `.local` is an expression, so your suggestion might be confusing from that respect?

Reply:
Doesn't seem like it to me. An expression is not a declaration; it can appear in a declaration, but not every term that an expression can appear in is a declaration. ("Term" = "subtree of the data model".)

Alternately, the text could begin with something like "Only a `.local` or an `.input` _declaration_ can change the meaning of a _variable_." Though this isn't quite true, as free variables introduce what we call "implicit declarations" someplace. A `.match` can't contain declarations (not even implicit declarations, since the only place where an "implicit declaration" makes sense is in a context where further names can be introduced) so it's clear from that that nothing within a `.match` changes the meaning of any variable name. 

Every phrasing I can think of is made more complicated by implicit declarations, though.

Reply:
@catamorphism Would you object to my merging the text as is? We're definitely coming back to the whole "resolved value" space later 😄

Reply:
@aphillips I don't object!